### PR TITLE
Real time simulation

### PR DESCRIPTION
**Pull request by lqy0057**
## What changes do you make in this PR?

#211 
* Please describe why you create this PR

## Checklist

* [x] I have merged the latest main branch into current branch.
* [x] I have run `bash scripts/format.sh` before merging.
* Please use "squash and merge" mode.

### CODE REVIEW

File: pgdrive/envs/pgdrive_env.py [hunk 1]
```diff
@@ -194,7 +193,7 @@ def step(self, action: np.ndarray):
             # traffic vehicles step
             self.scene_manager.step(self.pg_world.pg_config["physics_world_step_size"])
             self.pg_world.step()
-            if self.use_render and self.config["insert_frame"] and i < self.config["decision_repeat"] - 1:
+            if self.pg_world.force_fps(i < self.config["decision_repeat"] - 1):
                 # insert frame to render in min step_size
                 self.pg_world.taskMgr.step()
```
Comment (pengzhenghao) [inline pgdrive/envs/pgdrive_env.py:198]: This line still working when render is off?

File: pgdrive/world/force_fps.py [hunk 2]
```diff
@@ -20,20 +21,25 @@ def __init__(self, fps, start=False):
...
+            self.pg_world.taskMgr.remove("force_fps")
             self.state = self.UNLIMITED
             self.fps = None
+
+    def force_fps_task(self, task):
+        self.tick()
+        return task.cont
+
+    def __call__(self, other_condition):
+        return self.state == self.FORCED and self.pg_world.mode == RENDER_MODE_ONSCREEN and other_condition
```
Comment (pengzhenghao) [inline pgdrive/world/force_fps.py:45]: Stop using __call__ if you only mean for returning a boolean flag.

Create a more informative function name for this.

**Review by pengzhenghao — CHANGES_REQUESTED** [review]

File: pgdrive/world/force_fps.py (hunk 2 above)
Comment (lqy0057) [reply pgdrive/world/force_fps.py:45]: I will resolve this

**Comment by pengzhenghao** [comment]
What is the minimal script I need to use to verify the effectiveness of this PR?

**Comment by lqy0057** [comment]
> What is the minimal script I need to use to verify the effectiveness of this PR?

Every script, since the force fps function is switched to do real simulation. Press 'f' to experience

**Comment by lqy0057** [comment]
close #211

**Pull request merged**